src/Chainweb/Pact/Backend/RelationalCheckpointer.hs

```diff
@@ -296,19 +296,39 @@ doRegisterSuccessful :: Db -> PactHash -> IO ()
 doRegisterSuccessful dbenv (TypedHash hash) =
     runBlockEnv dbenv (indexPactTransaction $ BS.fromShort hash)
 
+doLookupSuccessful :: Db -> Maybe ConfirmationDepth -> V.Vector PactHash -> IO (HashMap.HashMap PactHash (T2 BlockHeight BlockHash))
+doLookupSuccessful dbenv confDepth hashes = runBlockEnv dbenv $ do
+    r <- callDb "doLookupSuccessful" $ \db -> do
+      let
+        currentHeightQ = "SELECT blockheight FROM BlockHistory \
+            \ ORDER BY blockheight DESC LIMIT 1"
 
-doLookupSuccessful :: Db -> PactHash -> IO (Maybe (T2 BlockHeight BlockHash))
-doLookupSuccessful dbenv (TypedHash hash) = runBlockEnv dbenv $ do
-    r <- callDb "doLookupSuccessful" $ \db ->
-         qry db qtext [ SBlob (BS.fromShort hash) ] [RInt, RBlob] >>= mapM go
-    case r of
-        [] -> return Nothing
-        (!o:_) -> return (Just o)
+      -- if there is a confirmation depth, we get the current height and calculate
+      -- the block height, to look for the transactions in range [0, current block height - confirmation depth]
+      blockheight <- case confDepth of
+        Nothing -> pure Nothing
+        Just (ConfirmationDepth cd) -> do
+          currentHeight <- qry_ db currentHeightQ [RInt]
+          case currentHeight of
+            [[SInt bh]] -> pure $ Just (bh - fromIntegral cd)
+            _ -> fail "impossible"
+
+      let
+        blockheightval = maybe [] (\bh -> [SInt bh]) blockheight
+        qvals = [ SBlob (BS.fromShort hash) | (TypedHash hash) <- V.toList hashes ] ++ blockheightval
+        qtypes = RInt : [ RBlob | _ <- V.toList hashes ] ++ maybe [] (const [RInt]) blockheight
```

Comment:
Maybe I'm missing something here but to me, this should be `[RInt, RBlob]` still.
You have `SELECT blockheight, hash` as the body of the query, that means the rows you get out are each a pair of a block height (int) and a hash (blob). And to me this list is supposed to be the list of the columns that are expected in each row returned by the query.

Reply:
Yep, my bad, I've interpreted this parameter as a list of types of the query arguments, but it's the list of types of expected results. Thanks!